app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/CommunityFragment.kt

```diff
@@ -1,8 +1,258 @@
 package com.epfl.drawyourpath.mainpage.fragments
 
+import android.os.Bundle
+import android.view.View
+import android.widget.ImageButton
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.drawerlayout.widget.DrawerLayout
 import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.community.Tournament
+import com.epfl.drawyourpath.community.CommunityTournamentPostViewAdapter
+import com.epfl.drawyourpath.community.TournamentModel
+import com.epfl.drawyourpath.community.TournamentPost
+import com.epfl.drawyourpath.path.Path
+import com.github.drawyourpath.bootcamp.path.Run
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.material.navigation.NavigationView
+import java.time.LocalDateTime
 
+/**
+ * fragment used to display and vote for the tournament posts [TournamentPost]
+ * TODO make it asynchronous when linked with the database
+ */
 class CommunityFragment : Fragment(R.layout.fragment_community) {
-    // TODO: add community UI here (like rating/competition) and the ViewModel for this screen/fragment.
+
+    private lateinit var drawer: DrawerLayout
+    private lateinit var menuButton: ImageButton
+    private lateinit var detailsLayout: LinearLayout
+    private lateinit var tournamentsView: RecyclerView
+
+    private lateinit var tournament: TournamentModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```

Comment:
This function could be improved in readability by factoring it even more in smaller functions in my opinion.

Reply:
I refactored it, now it is easier to understand